libdiod/diod_log.c

```diff
@@ -209,7 +209,12 @@ _verr (int errnum, const char *fmt, va_list ap)
 {
     char buf[128];
     char errbuf[64];
+#if defined(__FreeBSD__) || ((_POSIX_C_SOURCE >= 200112L) && ! _GNU_SOURCE)
+    strerror_r (errnum, errbuf, sizeof (errbuf)); /* XSI version */
+    char *s = errbuf;
+#else
     char *s = strerror_r (errnum, errbuf, sizeof (errbuf)); /* GNU version */
+#endif
```

Comment:
Probably better to use the built-in autconf macros here

> Macro: AC_FUNC_STRERROR_R
If strerror_r is available, define HAVE_STRERROR_R, and if it is declared, define HAVE_DECL_STRERROR_R. If it returns a char * message, define STRERROR_R_CHAR_P; otherwise it returns an int error number. The Thread-Safe Functions option of Posix requires strerror_r to return int, but many systems (including, for example, version 2.2.4 of the GNU C Library) return a char * value that is not necessarily equal to the buffer argument.

https://www.gnu.org/software/autoconf/manual/autoconf-2.60/html_node/Particular-Functions.html

Reply:
Done!